st2common/st2common/util/action_db.py

```diff
@@ -163,6 +165,10 @@ def update_liveaction_status(status=None, result=None, context=None, end_timesta
 
     LOG.debug('Updated status for LiveAction object: %s', liveaction_db)
 
+    if publish and status != old_status:
+        LiveAction.publish_status(liveaction_db)
```

Comment:
usually when publishing changes to properties. It is better to publish both old and new.

I think there is room for generalization here i.e. instead of only dealing with state change you can have a property_change exchange. And publish property changes .. At this point you need to publish ..

{
'objectid': 'skdfhd'
'property_name': 'foo'
'old': x
'new': y
}

That would be a nice generalization. We don't have to tackle it now - just something to think about.